thrift/thrift0.13/src/main/java/com/linecorp/armeria/internal/client/thrift/THttpClientDelegate.java

```diff
@@ -149,11 +150,10 @@ public RpcResponse execute(ClientRequestContext ctx, RpcRequest call) {
                 Exceptions.throwUnsafely(t);
             }
 
-            final Endpoint endpoint = ctx.endpoint();
             final HttpRequest httpReq = HttpRequest.of(
                     RequestHeaders.builder(HttpMethod.POST, ctx.path())
                                   .scheme(ctx.sessionProtocol())
-                                  .authority(endpoint != null ? endpoint.authority() : "UNKNOWN")
+                                  .authority(getAuthority(ctx))
```

Comment:
Question) Is it possible to just remove this logic and rely on Armeria's default behavior?
As far as I could tell, I couldn't find any logic which actually uses this value 😅

Reply:
When `autoFillSchemeAndAuthority()` is called, `HttpRequest` is null and it is lazily created here. So there is no chance to apply the default logic. 
https://github.com/line/armeria/blob/81f048557cfa09499ac43e04b5048340a7824a63/core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java#L420-L423
I tried to find some tweaks but failed. I had no choice but to add similar logic.

Reply:
I see 😅   Sounds reasonable.

I'm wondering if we want to push the logic for filling authority to `HttpClientDelegate` (or at least doing one final authority fill at this timing) since:
1) Users may want modify the `authority` header for `additionalHeaders` in decorators, which may not be reflected in `endpoint.host`
2) Duplicate logic doesn't have to exist in multiple locations

e.g. in the following location: 
https://github.com/line/armeria/blob/f4781d52a34f874e9459994afd8de763c9608480/core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java#L60-L70

This doesn't have to be handled in this PR, but just wanted to know your opinon 🙏

Reply:
It is a good idea to change its behavior in a better way. However, it has another breaking change that would be out of the scope of this PR. 
I prefer to handle it in a separate PR so that users easily see the change.

Reply:
Let me add a TODO for this.

Reply:
> I prefer to handle it in a separate PR so that users easily see the change.

Sounds good to me 👍